tests/tokens/StakeNFT.t.sol

```diff
@@ -2610,6 +2610,50 @@ contract StakeNFTTest is DSTest {
         assertEq(madToken.balanceOf(address(stakeNFT)), 100 * ONE_MADTOKEN);
     }
 
+    function testCollectRewardsAfterMinting() public {
+        (StakeNFT stakeNFT, MadTokenMock madToken, , ) = getFixtureData();
+        UserAccount user = newUserAccount(madToken, stakeNFT);
+
+        madToken.transfer(address(user), 100 * ONE_MADTOKEN);
+
+        user.approve(address(stakeNFT), 100 * ONE_MADTOKEN);
+
+        uint256 tokenID = user.mint(100 * ONE_MADTOKEN);
+
+        assertPosition(
+            getCurrentPosition(stakeNFT, tokenID),
+            StakeNFT.Position(uint224(100 * ONE_MADTOKEN), 1, 1, 0, 0)
+        );
+
+        setBlockNumber(block.number+2);
+
+        user.collectEth(tokenID);
+
+        assertEq(madToken.balanceOf(address(user)), 0);
+        assertEq(madToken.balanceOf(address(stakeNFT)), 100 * ONE_MADTOKEN);
+    }
+
+    function testCollectRewardsAfterMintingTo() public {
+        (StakeNFT stakeNFT, MadTokenMock madToken, , ) = getFixtureData();
+        UserAccount user = newUserAccount(madToken, stakeNFT);
+
+        madToken.approve(address(stakeNFT), 100 * ONE_MADTOKEN);
+
+        uint256 tokenID = stakeNFT.mintTo(address(user), 100 * ONE_MADTOKEN, 0);
```

Comment:
Suggestion: put a value different from 0 for lock the NFT for burnings. If the code is working withdraw should work just fine.